tests/regression/test_mesh_vertexonly.py

```diff
@@ -0,0 +1,333 @@
+def test_pic_swarm_in_plex_2d_3procs():
+    test_pic_swarm_in_plex(UnitSquareMesh(1, 1))
+
+
+# Mesh Generation Tests
+
+def verify_vertexonly_mesh(m, vm, inputvertexcoords, gdim):
+    """
+    Check that VertexOnlyMesh `vm` immersed in parent mesh `v` with
+    creation coordinates `inputvertexcoords` and geometric dimension
+    `gdim` behaves as expected. `inputvertexcoords` should be the same
+    for all MPI ranks to avoid hanging.
+    """
+    assert m.geometric_dimension() == gdim
+    # Correct dims
+    assert vm.geometric_dimension() == gdim
+    assert vm.topological_dimension() == 0
+    # Can initialise
+    vm.init()
+    # Find in-bounds and non-halo-region input coordinates
+    in_bounds = []
+    core, owned, ghost = m.cell_set.sizes
+    for i in range(len(inputvertexcoords)):
+        cell_num = m.locate_cell(inputvertexcoords[i])
+        if cell_num is not None and cell_num < owned:
+            in_bounds.append(i)
+    # Correct coordinates (though not guaranteed to be in same order)
+    assert np.shape(vm.coordinates.dat.data_ro) == np.shape(inputvertexcoords[in_bounds])
+    assert np.all(np.isin(inputvertexcoords[in_bounds], vm.coordinates.dat.data_ro))
+    # Correct parent topology
+    assert vm._parent_mesh is m
+    assert vm.topology._parent_mesh is m.topology
+    # Correct generic cell properties
+    assert np.shape(vm.cell_closure) == (len(inputvertexcoords[in_bounds]), 1)
+    with pytest.raises(AttributeError):
+        vm.cell_to_facets
+    assert vm.num_cells() == len(inputvertexcoords[in_bounds]) == vm.cell_set.size
+    assert vm.num_facets() == 0
+    assert vm.num_faces() == vm.num_entities(2) == 0
+    assert vm.num_edges() == vm.num_entities(1) == 0
+    assert vm.num_vertices() == vm.num_entities(0) == vm.num_cells()
+    # Correct parent cell numbers
+    stored_vertex_coords = np.copy(vm._swarm.getField("DMSwarmPIC_coor")).reshape((vm.num_cells(), gdim))
+    vm._swarm.restoreField("DMSwarmPIC_coor")
+    stored_parent_cell_nums = np.copy(vm._swarm.getField("parentcellnum"))
+    vm._swarm.restoreField("parentcellnum")
+    assert len(stored_vertex_coords) == len(stored_parent_cell_nums)
+    for i in range(len(stored_vertex_coords)):
+        assert m.locate_cell(stored_vertex_coords[i]) == stored_parent_cell_nums[i]
+
+
+@pytest.mark.parametrize("parentmesh", parentmeshes)
+def test_generate_cell_midpoints(parentmesh):
+    """
+    Generate cell midpoints for mesh m and check they lie in the correct cells
+    """
+    inputcoords, inputcoordslocal = cell_midpoints(parentmesh)
+    vm = VertexOnlyMesh(parentmesh, inputcoords)
+    # Midpoints located in correct cells of parent mesh
+    V = VectorFunctionSpace(parentmesh, "DG", 0)
+    f = Function(V).interpolate(parentmesh.coordinates)
+    # Check size of biggest len(vm.coordinates.dat.data_ro) so
+    # locate_cell can be called on every processor
+    max_len = MPI.COMM_WORLD.allreduce(len(vm.coordinates.dat.data_ro), op=MPI.SUM)
```

Comment:
This is the global size of the coordinates, not the max local size. You want `op=MPI.MAX` for that.

Reply:
Oops... Great spot. Thanks.